contracts/Controller.sol

```diff
@@ -75,6 +75,17 @@ contract Controller is ReentrancyGuard, Ownable {
         _;
     }
 
+    /**
+     * @notice modifier to check the validity of a specific vault id
+     * @param _accountOwner account owner address
+     * @param _vaultId vault id
+     */
+    modifier runPreprocessing(address _accountOwner, uint256 _vaultId) {
```

Comment:
I like the idea of using modifier here, but the name `runPreprocessing ` sounds like an internal function instead of a modifier.
Maybe renane to: `onlyOnValidVault`?

``